### PR TITLE
dot/syncer, tests: add execute_block call to syncer, fix storage sync tests

### PR DESCRIPTION
**Pull request by noot**
## Changes

- add execute_block call to syncer, fix block encoding
- check for storage value in all nodes in storage integration test

## Tests:

```
HOSTNAME=0.0.0.0 GOSSAMER_INTEGRATION_TEST_MODE=stress go test ./tests/stress/... -timeout=5m -p 1 -short -v -run TestStress_StorageChange
```

try with different `NETWORK_SIZE=?` values, I've had success with up to 6 nodes

## Checklist:

- [x] I have read [CONTRIBUTING](CONTRIBUTING.md) and [CODE_OF_CONDUCT](CODE_OF_CONDUCT.md)
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [x] I have linked any relevant issues in my pull request comments
- [x] All integration tests and required coverage checks are passing

## Issues:

- #852

### CODE REVIEW

**Comment by noot** [comment]
@edwardmack I've updated the test timing and added a PendingExtrinsic check, could you please pull and try again and let me know if it works better?

**Comment by edwardmack** [comment]
@noot, still failing for me 7 out of 10 tries, 6 of the error are the "did not find extrinsic in block on node ..." and one with chain heads not matching.

**Comment by noot** [comment]
@edwardmack I updated the test to check for the extrinsic in every node, I think it's working better now, passes every time for me locally. let me know how it works now

**Comment by edwardmack** [comment]
@noot, it's running way better, passed 7 out of 10, the 3 that failed had an error I haven't seen before:
```
--- FAIL: TestStress_StorageChange (18.39s)
    stress_test.go:287: submitted transaction to node charlie
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x7be468]

goroutine 20 [running]:
testing.tRunner.func1(0xc000172100)
        /usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x83d800, 0xdc9090)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/ChainSafe/gossamer/tests/stress.submitExtrinsicAssertInclusion(0xc000172100, 0xc0001a82e0, 0x3, 0x4, 0x94c720, 0xc0000a0ca0)
        /home/emack/projects/chainsafe/gossamer/tests/stress/stress_test.go:334 +0xed8
github.com/ChainSafe/gossamer/tests/stress.TestStress_StorageChange(0xc000172100)
        /home/emack/projects/chainsafe/gossamer/tests/stress/stress_test.go:377 +0x256
testing.tRunner(0xc000172100, 0x8d1e68)
        /usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:960 +0x350
FAIL    github.com/ChainSafe/gossamer/tests/stress      18.415s
FAIL
```

**Comment by noot** [comment]
@edwardmack ok awesome! I just pushed what should fix that error

**Comment by edwardmack** [comment]
@noot I'm getting a different error, I got this 2 times after running test 10 times:
```
--- FAIL: TestStress_StorageChange (28.45s)
    stress_test.go:287: submitted transaction to node charlie
    stress_test.go:392: got storage value from node alice: [0 0 115 104 101 114 101]
    stress_test.go:389: could not get storage value from node bob
    stress_test.go:392: got storage value from node charlie: [0 0 115 104 101 114 101]
    stress_test.go:396: 
                Error Trace:    stress_test.go:396
                Error:          Not equal: 
                                expected: 0
                                actual  : 1
                Test:           TestStress_StorageChange
                Messages:       [could not get storage value from node bob
                                ]
FAIL
FAIL    github.com/ChainSafe/gossamer/tests/stress      28.466s
FAIL
```

**Comment by noot** [comment]
@edwardmack that's probably due to the nodes being synced up with the extrinsic block at that point, I can try to add another sync check before then

File: dot/rpc/modules/chain.go [hunk 1]
```diff
@@ -236,12 +235,12 @@ func HeaderToJSON(header types.Header) ChainBlockHeaderResponse {
 		Digest:         ChainBlockHeaderDigest{},
 	}
 	if header.Number.Int64() == 0 {
-		res.Number = "0x0"
+		res.Number = "0x00"
```
Comment (ryanchristo) [inline dot/rpc/modules/chain.go:238]: maybe a comment to explain `0x00`

File: dot/types/block.go [hunk 2]
```diff
@@ -49,6 +49,10 @@ func (b *Block) Encode() ([]byte, error) {
 		return nil, err
 	}
 
+	if len(b.Header.Digest) == 0 {
```
Comment (ryanchristo) [inline dot/types/block.go:52]: maybe a comment here to explain

File: lib/babe/build.go [hunk 3]
```diff
@@ -31,6 +31,13 @@ import (
 	log "github.com/ChainSafe/log15"
 )
 
+// BuildBlock builds a block for the slot with the given parent.
+// TODO: separate block builder logic into separate module. The only reason this is exported is so other packages
```
Comment (ryanchristo) [inline lib/babe/build.go:35]: issue number?

File: tests/stress/stress_test.go [hunk 4]
```diff
@@ -344,24 +376,24 @@ func TestStress_StorageChange(t *testing.T) {
...
 		res := getStorage(t, node, key)
 
-		// TODO: currently, around 2/3 nodes have the updated state, even if they all have the same
-		// chain head. figure out why this is the case and fix it.
-		idx := rand.Intn(len(nodes))
-		if idx == node.Idx {
-			// TODO: why does finalize_block modify the storage value?
-			require.NotEqual(t, []byte{}, res)
-			require.Equal(t, true, bytes.Contains(value, res[2:]))
+		// TODO: why does finalize_block modify the storage value?
```
Comment (ryanchristo) [inline tests/stress/stress_test.go:387]: issue number?

Comment (noot) [reply tests/stress/stress_test.go:387]: #851

File: lib/babe/build.go (hunk 3 above)
Comment (noot) [reply lib/babe/build.go:35]: will open an issue

Comment (noot) [reply lib/babe/build.go:35]: #882

**Pull request merged**